go/types/type_cache.go

```diff
@@ -537,6 +537,10 @@ func MakeUnionType(elemTypes ...*Type) *Type {
 	return makeSimplifiedUnion(elemTypes...)
 }
 
+func MakeFullUnionType(elemTypes ...*Type) *Type {
```

Comment:
Agreed. I was reluctant to mess with the existing terminology cause I was trying to minimize the change.